src/veh_type.cpp

```diff
@@ -326,8 +365,65 @@ void vpart_info::finalize()
 
 void vpart_info::check()
 {
-    for( auto &part_ptr : vehicle_part_int_types ) {
-        auto &part = *part_ptr;
+    for( auto &vp : vehicle_part_types ) {
+        auto &part = vp.second;
+
+        // handle legacy parts without requirement data
+        if( part.legacy ) {
+            part.install_reqs.components = { { { { part.item, 1 } } } };
+
+            part.install_skills.emplace( skill_mechanics, part.difficulty );
+            part.removal_skills.emplace( skill_mechanics, part.difficulty );
```

Comment:
2 is fine for now. Later on it could be something like `(install difficulty - 2)` or so.